includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -404,7 +404,31 @@ public function admin_options() {
 		// Add notices to the WooCommerce Payments settings page.
 		do_action( 'woocommerce_woocommerce_payments_admin_notices' );
 
-		parent::admin_options();
+		if ( ! WC_Payments_Admin::is_grouped_settings_enabled() ) {
```

Comment:
It looks like this is creating a circular dependency.
`WC_Payments_Admin` includes and consumes this class. But now this class is also consuming `WC_Payments_Admin`.

Can you please ensure that there isn't a circular dependency? I would be fine even having the line `get_option( '_wcpay_feature_grouped_settings', '0' ) === '1'` duplicated in here 🤷 
Otherwise maybe it could be an extra argument in the `WC_Payment_Gateway_WCPay` constructor (but that's an extra argument, which is not too good) or the whole `get_option( '_wcpay_feature_grouped_settings', '0' ) === '1'` could be extracted to a separate class and a separate method 🤷

Reply:
First, thanks for pointing this out, I wasn't aware there was an instance of `WC_Payment_Gateway_WCPay` in `WC_Payments_Admin`. I've been checking the logs and looks like this isn't causing any trouble. I'm not a PHP expert so I'm not 100% sure about this, but I've been investigating and apparently this wouldn't be a circular dependency because `WC_Payment_Gateway_WCPay` is accessing `WC_Payments_Admin` statically.

Did you find any errors in your logs while testing this PR??

Reply:
No, I didn't see any errors in the logs. Not all the circular references create errors in logs, but they can create issues when refactoring or when trying to test.
As I'm re-evaluating, maybe this instance doesn't seem to be a big deal, considering that the feature flag will be hopefully short lived 👍

Reply:
I was asking about the errors in the logs because I couldn't get any runtime errors while executing the code containing the circular references, and sounded like you did find runtime errors. Thanks for letting me know!
Ok, sounds good to me too!